### PR TITLE
Add note to dev cheat sheet on tests and public(friend) functions

### PR DESCRIPTION
**Pull request by rockbmb**
## Description 

This PR adds a note to the Sui Dev cheat sheet on testing private functions outside their module via `public(friend)`.

## Test Plan 

None required.

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: doc/src/build/dev_cheat_sheet.md [hunk 1]
```diff
@@ -32,6 +32,31 @@ Quick reference on best practices for Sui Network developers.
...
+```move
+module package::mod {
+    friend package::test_mod;
+
+    public(friend) fun foo() {}
+}
+```
+
+As mentioned above, the signatures of `public(friend)` functions can be changed at will in future package versions.
+Note that, in this case, commands such as `sui move build`/`sui move coverage` will need the addition of the `--test` flag.
```
Comment (ronny-mysten) [inline doc/src/build/dev_cheat_sheet.md:58]:
```suggestion
When using commands such as `sui move build` and `sui move coverage` with these modules, you must include the `--test` flag. See [Build and Test the Sui Move Package](move/build-test.md#building-your-package) for more information on building packages.
```

**Review by ronny-mysten — COMMENTED** [review]
Notes to conform with style guide.

**Comment by rockbmb** [comment]
@ronny-mysten thanks for the tips! Fixed.

**Comment by rockbmb** [comment]
Hello @ronny-mysten,

I apologize for bothering you with this very minor PR - would it still be possible to merge this?

Thanks!

**Comment by ronny-mysten** [comment]
Apologies @rockbmb! There is no such thing as too minor a PR that you could "bother" me with. I just thought one of the devs had reviewed. My bad!

**Review by ronny-mysten — APPROVED** [review]

**Pull request merged**